src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -79,16 +67,16 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
     /**
      * Deletes prescriptions that are past the end date and stores them in the completed prescription list.
      */
-    public void checkAndMoveEndedPrescriptions() throws IOException {
+    public void checkAndMoveEndedPrescriptions() {
         PrescriptionList prescriptionListCopy = new PrescriptionList(model.getPrescriptionList());
         for (Prescription prescription : prescriptionListCopy.getPrescriptionList()) {
             if (prescription.isEnded()) {
                 model.deletePrescription(prescription);
+            }
+            if (!model.hasCompletedPrescription(prescription)) {
                 model.addCompletedPrescription(prescription);
```

Comment:
this part causing tests to fail.

Doing this can cause a medication to end up in both the normal and completed list.

Should be isEnded **and** not in completed list.

Reply:
lgtm plz